common/static/common/js/components/views/feedback.js

```diff
@@ -0,0 +1,151 @@
+;(function (define) {
+    'use strict';
+    define(["jquery", "underscore", "underscore.string", "backbone", "common/js/components/utils/templates",
+        'text!common/templates/components/system-feedback.underscore'],
+    function($, _, str, Backbone, TemplateUtils, system_feedback_template) {
+        var SystemFeedback = Backbone.View.extend({
+            options: {
+                title: "",
+                message: "",
+                intent: null,  // "warning", "confirmation", "error", "announcement", "step-required", etc
+                type: null, // "alert", "notification", or "prompt": set by subclass
+                shown: true,  // is this view currently being shown?
+                icon: true,  // should we render an icon related to the message intent?
+                closeIcon: true,  // should we render a close button in the top right corner?
+                minShown: 0,  // length of time after this view has been shown before it can be hidden (milliseconds)
+                maxShown: Infinity  // length of time after this view has been shown before it will be automatically hidden (milliseconds)
+
+            /* Could also have an "actions" hash: here is an example demonstrating
+               the expected structure. For each action, by default the framework
+               will call preventDefault on the click event before the function is
+               run; to make it not do that, just pass `preventDefault: false` in
+               the action object.
+
+            actions: {
+                primary: {
+                    "text": "Save",
+                    "class": "action-save",
+                    "click": function(view) {
+                        // do something when Save is clicked
+                    }
+                },
+                secondary: [
+                    {
+                        "text": "Cancel",
+                        "class": "action-cancel",
+                        "click": function(view) {}
+                    }, {
+                        "text": "Discard Changes",
+                        "class": "action-discard",
+                        "click": function(view) {}
+                    }
+                ]
+            }
+            */
+            },
+
+            initialize: function() {
+                if (!this.options.type) {
+                    throw "SystemFeedback: type required (given " +
+                        JSON.stringify(this.options) + ")";
+                }
+                if (!this.options.intent) {
+                    throw "SystemFeedback: intent required (given " +
+                        JSON.stringify(this.options) + ")";
+                }
+                //this.template = TemplateUtils.loadTemplate("system-feedback");
+                this.template = _.template(system_feedback_template);
```

Comment:
Nit: I've taken to not caching the template on the object. Most classes only render once so it just complicates the code unnecessarily.